docs/source/tutorial_mor_with_anns.rst

```diff
@@ -0,0 +1,271 @@
+automatically train several neural networks with different initial weights and
+select the one leading to the best results on the validation set. We can also
+set `ann_mse` to `'like_basis'`. Then, the algorithm tries to train a neural
+network that leads to a mean squared error on the training set that is as small
+as the error of the reduced basis. If the maximal number of restarts is reached
+without finding a network that fulfills the tolerances, an exception is raised.
+In such a case, one could try to change the architecture of the neural network
+or switch to `ann_mse=None` which is guaranteed to produce a reduced order
+model (perhaps with insufficient approximation properties).
+
+We can now construct a reductor with prescribed error for the basis and mean
+squared error of the neural network:
+
+.. jupyter-execute::
+
+    from pymor.reductors.neural_network import NeuralNetworkReductor
+
+    reductor = NeuralNetworkReductor(fom,
+                                     training_set,
+                                     validation_set,
+                                     l2_err=1e-5,
+                                     ann_mse=1e-5)
+
+To reduce the model, i.e. compute a reduced basis via POD and train the neural
+network, we use the respective function of the
+:class:`~pymor.reductors.neural_network.NeuralNetworkReductor`:
+
+.. jupyter-execute::
+
+    rom = reductor.reduce()
+
+We are now ready to test our reduced model by solving for a random parameter value
+the full problem and the reduced model and visualize the result:
+
+.. jupyter-execute::
+
+    mu = parameter_space.sample_randomly(1)[0]
+
+    U = fom.solve(mu)
+    U_red = rom.solve(mu)
+    U_red_recon = reductor.reconstruct(U_red)
+
+    fom.visualize((U, U_red_recon),
+                  legend=(f'Full solution for parameter {mu}', f'Reduced solution for parameter {mu}'))
+
+Finally, we measure the error of our neural network and the performance
+compared to the solution of the full order problem on a training set. To this
+end, we sample randomly some |parameter values| from our |ParameterSpace|:
+
+.. jupyter-execute::
+
+    test_set = parameter_space.sample_randomly(10)
+
+Next, we create empty solution arrays for the full and reduced solutions and an
+empty list for the speedups:
+
+.. jupyter-execute::
+
+    U = fom.solution_space.empty(reserve=len(test_set))
+    U_red = fom.solution_space.empty(reserve=len(test_set))
+
+    speedups = []
+
+Now, we iterate over the test set, compute full and reduced solutions to the
+respective parameters and measure the speedup:
+
+.. jupyter-execute::
+
+    import time
+
+    for mu in test_set:
+        tic = time.time()
+        U.append(fom.solve(mu))
+        time_fom = time.time() - tic
+
+        tic = time.time()
+        U_red.append(reductor.reconstruct(rom.solve(mu)))
+        time_red = time.time() - tic
+
+        speedups.append(time_fom / time_red)
```

Comment:
It should be more accurate to use `time.perf_counter` instead of `time.time`.

Reply:
Pretty much all the demos use `time.time` though atm. Also `time.process_time` is IMO the better choice since that does not include sleep. See also src/pymor/tools/timing.py

Reply:
Then I will leave as is and we could think about changing the demos all at once.